src/platform/middleware.ts

```diff
@@ -73,3 +75,48 @@ export const gracefullyDenyDeletedDevices: RequestHandler = (
 			next();
 		});
 };
+
+// register the device state change handler which updates the device model...
+_.once(() => {
+	DeviceOnlineState.getDefaultManager().on(
+		'change',
+		(uuid: string, newState: DeviceOnlineState.DeviceOnlineStates) => {
+			if (!uuid) return;
+
+			// patch the api_heartbeat_state value to the new state...
+			resinApi.patch({
+				resource: 'device',
+				options: {
+					$filter: {
+						uuid,
+						api_heartbeat_state: {
+							$ne: newState,
```

Comment:
This should mean that an UPDATE query is run, but no rows will match if the device has the same state. It would be good to get @Page- views on this, and if we should make it a SELECT/UPDATE instead.

Reply:
Speaking to Petros, this approach seems the most logical, so unless you have any concerns @Page- then I suggest we stick with it :+1: